homeassistant/components/influxdb.py

```diff
@@ -33,8 +35,8 @@
 CONFIG_SCHEMA = vol.Schema({
     DOMAIN: vol.Schema({
         vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
-        vol.Required(CONF_USERNAME): cv.string,
-        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_USERNAME, default=DEFAULT_USERNAME): cv.string,
+        vol.Optional(CONF_PASSWORD, default=DEFAULT_PASSWORD): cv.string,
```

Comment:
I just tested it, and in the default InfluxDB configuration, one can supply only username and only password, and it will still work, because Influxdb simply ignores them.

When authentication is enabled, however, one can create username with password that's a blank string, but a blank string is still required to be passed, so it makes sense to require password's presence if username is given.

I will update